--- a/drivers/usb/gadget/function/f_mtp.c
+++ b/drivers/usb/gadget/function/f_mtp.c
@@ -1421,12 +1421,6 @@
 	dev->cdev = cdev;
 	mtp_log("dev: %pK\n", dev);
 
-	/* ChipIdea controller supports 16K request length for IN endpoint */
-	if (cdev->gadget->is_chipidea && mtp_tx_req_len > 16384) {
-		mtp_log("Truncating Tx Req length to 16K for ChipIdea\n");
-		mtp_tx_req_len = 16384;
-	}
-
 	/* allocate interface ID(s) */
 	id = usb_interface_id(c, f);
 	if (id < 0)
