diff --git a/drivers/input/touchscreen/goodix.c b/drivers/input/touchscreen/goodix.c
index f8798d11ec03..d860a1c63b73 100644
--- a/drivers/input/touchscreen/goodix.c
+++ b/drivers/input/touchscreen/goodix.c
@@ -1190,13 +1190,16 @@ static int goodix_configure_dev(struct goodix_ts_data *ts)
 		dev_err(&ts->client->dev,
 			"Invalid config (%d, %d, %d), using defaults\n",
 			ts->prop.max_x, ts->prop.max_y, ts->max_touch_num);
-		ts->prop.max_x = GOODIX_MAX_WIDTH - 1;
-		ts->prop.max_y = GOODIX_MAX_HEIGHT - 1;
-		ts->max_touch_num = GOODIX_MAX_CONTACTS;
-		input_abs_set_max(ts->input_dev,
-				  ABS_MT_POSITION_X, ts->prop.max_x);
-		input_abs_set_max(ts->input_dev,
-				  ABS_MT_POSITION_Y, ts->prop.max_y);
+		if (!ts->prop.max_x || !ts->prop.max_y) {
+			ts->prop.max_x = GOODIX_MAX_WIDTH - 1;
+			ts->prop.max_y = GOODIX_MAX_HEIGHT - 1;
+			input_abs_set_max(ts->input_dev,
+					  ABS_MT_POSITION_X, ts->prop.max_x);
+			input_abs_set_max(ts->input_dev,
+					  ABS_MT_POSITION_Y, ts->prop.max_y);
+		}
+		if (!ts->max_touch_num)
+			ts->max_touch_num = GOODIX_MAX_CONTACTS;
 	}
 
 	if (dmi_check_system(nine_bytes_report)) {
diff --git a/drivers/ufs/host/ufs-qcom.c b/drivers/ufs/host/ufs-qcom.c
index 9c0973a7ffc3..40f0ef83deb5 100644
--- a/drivers/ufs/host/ufs-qcom.c
+++ b/drivers/ufs/host/ufs-qcom.c
@@ -2414,10 +2414,10 @@ static int ufs_qcom_config_esi(struct ufs_hba *hba)
 		qi[idx].irq = msi_get_virq(hba->dev, idx);
 		qi[idx].idx = idx;
 		qi[idx].hba = hba;
 
-		ret = devm_request_irq(hba->dev, qi[idx].irq, ufs_qcom_mcq_esi_handler,
-				       IRQF_SHARED, "qcom-mcq-esi", qi + idx);
+		ret = devm_request_threaded_irq(hba->dev, qi[idx].irq, NULL, ufs_qcom_mcq_esi_handler,
+				       IRQF_ONESHOT | IRQF_SHARED, "qcom-mcq-esi", qi + idx);
 		if (ret) {
 			dev_err(hba->dev, "%s: Failed to request IRQ for %d, err = %d\n",
 				__func__, qi[idx].irq, ret);
 			/* Free previously allocated IRQs */
